network/dag/bbolt_dag.go

```diff
@@ -289,7 +289,7 @@ func (dag *bboltDAG) add(transaction Transaction) error {
 				return fmt.Errorf("unable to store forward reference %s->%s: %w", prev, ref, err)
 			}
 			if !exists(transactions, prev) {
-				log.Logger().Debugf("Transaction %s is referring to missing prev %s, marking it as missing", ref, prev)
+				log.Logger().Debugf("Transaction is referring to missing prev, marking it as missing (tx=%s, prev=%s)", ref, prev)
```

Comment:
Perhaps use the `WithFields` as described here in the logrus docs;
https://github.com/sirupsen/logrus#fields

Reply:
I still don't agree; I'd use fields for key-value pairs that are applicable to all (or most) of the log entries, e.g. remote client IP address, authenticated user, module name (one of our current) since that would make it very useful for log analytics (e.g. using ELK). If you have a myriad of very different fields, it makes the log (and thus analytics) messy in my opinion.